### DISCUSSION TITLE
Climate MQTT Mode Set

### DISCUSSION THREAD

**Discussion by fiorentino77** [indomus/forum#317] [GENERALE - Discussioni generiche]
Salve a tutti,
ho creato un'entita' CLIMATE tramite comandi MQTT e Node-Red.
Il codice e' il seguente:

```
> - platform: mqtt
>   name: "Termo Taverna"
>   retain: false
>   qos: 0
>   modes:
>     - heat
>     - cool
>     - auto
>     - 'off'
>   mode_state_topic: "termo/taverna/modestate"
>   mode_state_template: >-
>     {% set values = { 'true':'heat', 'false':'off'} %}
>     {{ values[value] if value in values.keys() else 'off' }}
>   mode_command_topic: "termo/taverna/modestate/command"
>   mode_command_template: >-
>     {% set values = { 'heat':'0', 'cool':'1',  'auto':'2', 'off':'4'} %}
>     {{ values[value] if value in values.keys() else '4' }}
>   current_temperature_topic: "termo/taverna/temp"
>   temperature_state_topic: "termo/taverna/TempSet"
>   temperature_command_topic: "termo/taverna/setpoint"
>   min_temp: 13
>   max_temp: 25.5
>   temp_step: 0.1

```
Tutto funziona come dovrebbe ma solamente una cosa non riesco a sistemare.
Dato che Node-red risponde solo con true o false sul topic modeset e quest'ultimo ha invece 4 opzioni come fare il "parse" delle rimanenti due (cool e auto)?
Una soluzione potrebbe essere in optimistyc mode ma non fa al caso mio.
Spero di essere stato chiaro.

Saluti e grazie,

Alessandro

**Comment by snakuzzo** [comment]
Correggi il post utilizzando il markdown per il codice, altrimenti dovrò cancellare. Grazie
